### PR TITLE
Feature/loading button

### PR DESCRIPTION
**Pull request by phoward38**
Added `loading_button` & `reset_loading_button` functions.
- Spinner added to button while loading
- Custom text & CSS styling arguments for button before & during loading

Questions:
- Do I need to account for width & the text within the button (i.e. overflow)? Or should this be up to the user's styling discretion?

### CODE REVIEW

File: R/loading_button.R [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+                           label,
+                           style = "color: #FFFFFF; background-color: #337AB7; border-color: #2E6DA4;",
+                           loading_text = "Loading...",
+                           loading_style = "color: #FFFFFF; background-color: grey; border-color: grey;"
+) {
+  
+  htmltools::tagList(
+    tags$button(
+      id = paste0("loading_button-", id),
+      class = "btn style",
```
Comment (merlinoa) [inline R/loading_button.R]: I would replace the value for class here with an additional `class` argument to `loading_button()`

File: R/loading_button.R [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      label
+    ),
+    tagList(
+      shiny::singleton(
+        tags$head(
+          # Used personalized kit from FA
+          tags$script(
+            src = "https://kit.fontawesome.com/fb4611ff56.js"
+          ),
+          tags$style(
```
Comment (merlinoa) [inline R/loading_button.R]: remove this `tags$style` and pass the styles directly to the "style" arguments of the buttons.  The namespacing for the styles with multiple buttons would get messed up with this current implementation (i.e. there would be multiple "style" classes in the css and the buttons wouldn't necessarily use the correct style)

File: R/loading_button.R [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+              
+              .loading_style {",
+              loading_style,
+              "}
+              "
+            )
+          ),
+          # Script for disabling button & changing text
+          tags$script(
+            HTML(
```
Comment (merlinoa) [inline R/loading_button.R]: move this js to a new file named "inst/srcjs/loading_button.js".  Put this js in a js function named "loading_button" and source + call this function from the ui of this button.  I added a new `.onLoad()` function to tychobratools that will make all js files in "inst/srcjs" available on the frontend in a "tychobratools" folder; so you can now source in your new js file with `tags$script(src="tychobratools/loading_button.js")`.  Let me know if you have questions on how this works.

Also remove the `debugger`

**Review by merlinoa — CHANGES_REQUESTED** [review]
Please make changes per my comments on the code.  I added a new test app at "inst/examples/loading_button_test" that you can use for testing.  Let me know if you have questions or  think we should go another direction with any of my requested changes.

File: R/loading_button.R (hunk 1 above)
Comment (phoward38) [reply R/loading_button.R]: f6448b6003a856febc94fe254872f15ec9e79b1f

File: R/loading_button.R (hunk 2 above)
Comment (phoward38) [reply R/loading_button.R]: f6448b6003a856febc94fe254872f15ec9e79b1f

File: R/loading_button.R (hunk 3 above)
Comment (phoward38) [reply R/loading_button.R]: eca8b3944850fc1ee9d75ecab923de7341f18380

**Comment by phoward38** [comment]
@merlinoa Resolved & added commits for each

**Review by merlinoa — APPROVED** [review]

**Pull request merged**